### PR TITLE
This pull request adds elm support for the elm language server found at https://github.com/elm-tooling/elm-language server.

### PR DESCRIPTION
**Pull request by Daniel-V1**

### CODE REVIEW

**Comment by yyoncho** [comment]
Did you close it by mistake or you do not want to be merged as it is?

**Comment by Daniel-V1** [comment]
I was thinking maybe I should actually wait until they get the server separated into its own thing more easily downloaded, or until I can include a way to clone the repo automatically for users. Thoughts?

**Comment by yyoncho** [comment]
@Daniel-V1 not all servers have automatic/easy installation but still, there are users who use them. IMHO this PR will bring up value for the lsp-mode/elm users so I am fine with it as long as you populate the readme with pointers/instructions on how to install the server.

**Comment by Daniel-V1** [comment]
Do these instructions look ok? Should I do something to get rid of the duplicate commits?

File: lsp-elm.el [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  (locate-user-emacs-file "elm-language-server/")
+  "Install directory for elm-language-server.  A slash is expected at the end.
+This directory should contain a clone of elm-tooling/elm-language-server from github."
+  :group 'lsp-elm
+  :risky t
+  :type 'directory)
+
+(defun lsp-elm--elm-command ()
+  "Generate LSP startup command for the Elm Language Server."
+  `("node" ,(concat (file-truename lsp-elm-server-install-dir)
```
Comment (yyoncho) [inline lsp-elm.el]: Use f-join and remove the comment about `A slash is ...`. f-join will handle that.

File: README.org [hunk 2]
```diff
@@ -109,41 +109,42 @@
...
+   | Vue                   | [[https://github.com/vuejs/vetur/tree/master/server][vue-language-server]]                       | Yes            | npm install -g vue-language-server                                                                                                                       | Yes (Firefox/Chrome)         |
+   | XML                   | [[https://github.com/angelozerr/lsp4xml][lsp4xml]]                                   | Yes            | Download from [[https://github.com/angelozerr/lsp4xml/releases][lsp4xml releases]]                                                                                                                           |                              |
+   | Elm                   | [[https://github.com/elm-tooling/elm-language-server][elmLS]]                                     | Yes            | Clone the [[https://github.com/elm-tooling/elm-language-server][repository]], run npm install in the root, then run [[https://www.typescriptlang.org][tsc]] in the server directory. Set lsp-elm-server-install-dir to wherever you cloned the repo. | No                           |
```
Comment (yyoncho) [inline README.org]: Can you move it on the proper place (the languages are sorted alphabetically)?

File: lsp-elm.el (hunk 1 above)
Comment (Daniel-V1) [reply lsp-elm.el]: Should be good now

File: README.org (hunk 2 above)
Comment (Daniel-V1) [reply README.org]: Think this is in the right place now. Sorry I didn't even think about sorting lol

**Comment by yyoncho** [comment]
Thank you and congrats for your first `lsp-mode` PR! :tada:

**Comment by Daniel-V1** [comment]
Thank you so much for helping me get this in and for lsp mode and dap mode!

**Pull request merged**